client/src/modules/authorization/components/ResetPassword.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { PureComponent } from 'react';
+import validator from 'validator';
+import { FormattedMessage } from 'react-intl';
+
+class ResetPassword extends PureComponent {
+  state = {
+    email: '',
+    tipVisible: false
+  };
+
+  handleInputChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState({ email: value });
+  };
+
+  handleEmailValidation = () => {
+    const { email } = this.state;
+
+    return validator.isEmail(email);
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const isEmailCorrect = this.handleEmailValidation();
+
+    if (isEmailCorrect) {
+      // handle submit here
+      this.hideTip();
+
+      return;
+    }
+
+    this.showTip();
```

Comment:
Please reverse the if. Check if not ok then just show tip and return. Handling submittion would have probably more than one line.
```
if (!isEmailCorrect) {
  this.showTip();
  return;
}
...
```